ever_given/ever_given/engines/singularity_engine.py

```diff
@@ -0,0 +1,77 @@
+import io
+from pathlib import Path
+import subprocess
+import typing
+
+from .utils import ContainerInstance, Engine, GUEST_OUTPUT_DIR
+
+AFTER_TERMINATE_WAIT = 5  # Seconds to wait after 'terminate' signal before 'kill'
+
+
+class SingularityContainerInstance(ContainerInstance):
+    def __init__(self, process: subprocess.Popen):
+        self.process = process
+
+    def logs(self, *, want_stdout: bool, want_stderr: bool):
+        if want_stdout and want_stderr:
+            raise ValueError("Can't have want_stdout and want_stderr both true")
+        pipe: io.TextIOBase
+        if want_stdout:
+            pipe = typing.cast(io.TextIOBase, self.process.stdout)
+        elif want_stderr:
+            pipe = typing.cast(io.TextIOBase, self.process.stderr)
+        else:
+            raise ValueError("Can't have want_stdout and want_stderr both false")
+        return pipe.readline()
```

Comment:
I don't know if this works - would need to return a generator to line up with Docker's container